internal/middleware/middleware_test.go

```diff
@@ -0,0 +1,67 @@
+package middleware
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus/hooks/test"
+)
+
+func TestIDMiddleware(t *testing.T) {
+	req := httptest.NewRequest("GET", "/", nil)
+	logger, hook := test.NewNullLogger()
+	entry := logger.WithField("entry", "exists")
+
+	testRequest(t, req, IDMiddleware(entry))
+
+	if len(hook.Entries) != 1 {
+		t.Fatalf("error: len = 1")
+	}
+	t.Log(hook.LastEntry().Data["correlation_id"])
```

Comment:
there is no much sense to print correlation_id. You should assert that entry has that filed instead

Reply:
+